pkg/operator/ceph/cluster/controller.go

```diff
@@ -496,36 +481,13 @@ func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 
 	// if the image changed, we need to detect the new image version
 	if oldClust.Spec.CephVersion.Image != newClust.Spec.CephVersion.Image {
-		logger.Infof("the ceph version changed. detecting the new image version...")
-		version, err := cluster.detectCephVersion(c.rookImage, newClust.Spec.CephVersion.Image, detectCephVersionTimeout)
+		logger.Infof("the ceph version changed")
+		version, _, err := c.detectAndValidateCephVersion(cluster, newClust.Spec.CephVersion.Image)
```

Comment:
This function is only used twice, once with the middle parameter omitted and once with it kept. In this situation, having this function doesn't add value over simply choosing to call `validateCephVersion` after the latter situation and not calling it here.

Reply:
In both situations I need to call `validateCephVersion()`. The difference is whether or not the error means we should retry the operation. So I see it reducing the duplication by having the helper method.